packages/router-generator/src/generator.ts

```diff
@@ -312,9 +332,40 @@ export async function generator(config: Config) {
       return
     }
 
+    node.isVirtualParentRequired =
+      removeGroups(node.path ?? '')
+        .split('')
+        .filter((char) => char === '/').length >= 2 && // check that the parent route wouldn't be the root route
```

Comment:
can the "check that the parent route wouldn't be the root route " be simplified to this?

```tsx
const node.parent?.parent
```

Reply:
The problem is that this condition never comes to pass.

Since a nested layout route (with no `.route.tsx` configuration file) wouldn't ever be detected as having a parent route because the `routeNodes` array wouldn't already have a virtual parent route to link it to.

In the next chunk of code below is where a virtual parent route is created, the current route is added as its child, and the newly created parent node is mapped to the current node's `.parent` prop.

Reply:
There's an alternative that works.

```tsx
const cleanedPathIsEmpty = (node.cleanedPath || '').length === 0
node.isVirtualParentRequired = !cleanedPathIsEmpty && !node.parent && node.isLayout
```